autogluon/task/text_prediction/models/basic_v1.py

```diff
@@ -0,0 +1,911 @@
+                test_data = pd.DataFrame(test_data)
+            test_data = TabularDataset(test_data,
+                                       columns=self._feature_columns,
+                                       column_properties=self._column_properties)
+        processed_test = self._preprocessor.process_test(test_data)
+        inference_batch_size = self.config.optimization.per_device_batch_size\
+                               * self.config.optimization.val_batch_size_mult
+        test_dataloader = DataLoader(processed_test,
+                                     batch_size=inference_batch_size,
+                                     shuffle=False,
+                                     batchify_fn=self._preprocessor.batchify(is_test=True))
+        test_predictions = _classification_regression_predict(self._net,
+                                                              dataloader=test_dataloader,
+                                                              problem_type=self._problem_types[0],
+                                                              has_label=False)
+        if self._problem_types[0] == _C.CLASSIFICATION:
+            if get_probabilities:
+                return test_predictions
+            else:
+                test_predictions = test_predictions.argmax(axis=-1)
+                if get_original_labels:
+                    test_predictions = np.array(
+                        list(map(self._column_properties[self._label_columns[0]].inv_transform,
+                                 test_predictions)))
+        return test_predictions
+
+    def predict_proba(self, test_data):
+        """Predict with probability
+
+        Parameters
+        ----------
+        test_data
+            The test data. Can be a pandas DataFrame or a file containing a pandas dataframe
+
+        Returns
+        -------
+        probabilities
+            The probabilities. Shape (#Samples, num_class)
+        """
+        assert self.problem_types[0] == _C.CLASSIFICATION
+        return self._internal_predict(test_data,
+                                      get_original_labels=False,
+                                      get_probabilities=True)
+
+    def predict(self, test_data, get_original_labels=True):
+        """Get the prediction results
+
+        Parameters
+        ----------
+        test_data
+            tabular dataset
+        get_original_labels
+            Whether to get the original labels.
+            For example, the labels can be "entailment", "not_entailment" and whether
+            to get the original string labels or get the int values.
+
+        Returns
+        -------
+        predictions
+            The predictions. Shape (#Samples,)
+        """
+        return self._internal_predict(test_data,
+                                      get_original_labels=get_original_labels,
+                                      get_probabilities=False)
+
+    def save(self, dir_path):
```

Comment:
Add TODO to add support for saving/loading to S3 in future. Tabular has this functionality for many of its models.